### PR TITLE
Run dart-services with a 2.0 SDK

### PR DESCRIPTION
**Pull request by jcollins-g**
This does not yet change the version being used to compile/analyze dart code for Dartpad, it just changes the version of the SDK running the server-side software.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,4 +1,4 @@
-FROM google/dart-runtime:1.24.3
+FROM google/dart-runtime:2.0.0-dev.58.0
```
Comment (devoncarew) [inline Dockerfile]: Is there one for this? @athomas, do you know who releases the docker images?

File: example/index.dart [hunk 2]
```diff
@@ -8,13 +8,13 @@ import "dart:html";
...
     String code = querySelector('#code').text;
-    var json = {'source': code};
+    var jsonData = {'source': code};
     //var foo = querySelector('#apiEndPoint');
     String api = (querySelector('#apiEndPoint') as InputElement).value;
 
     Stopwatch sw = new Stopwatch()..start();
     HttpRequest
-        .request(api, method: 'POST', sendData: JSON.encode(json))
+        .request(api, method: 'POST', sendData: json.encode(jsonData))
```
Comment (devoncarew) [inline example/index.dart]: FYI, it's also possible to use `jsonEncode`, which let's you keep `json` as handy local variable names...

File: pubspec.yaml [hunk 3]
```diff
@@ -4,7 +4,7 @@ description: The server backend for a web based interactive Dart service.
 homepage: https://github.com/dart-lang/dart-services
 
 environment:
-  sdk: '>=1.0.0 <2.0.0'
+  sdk: '>=2.0.0-dev.54 <2.0.0'
```
Comment (devoncarew) [inline pubspec.yaml]: no real opinion, but perhaps `2.0.0-dev.58.0` to match above?

**Comment by devoncarew** [comment]
Ah, I see some travis failures: https://travis-ci.org/dart-lang/dart-services/builds/383412574#L639

File: pubspec.yaml (hunk 3 above)
Comment (jcollins-g) [reply pubspec.yaml]: 54 has the last batch of constant changes, which is why I set it to that.

File: Dockerfile (hunk 1 above)
Comment (jcollins-g) [reply Dockerfile]: It seemed to deploy alright, so I suspect there is a docker image built.

**Comment by devoncarew** [comment]
Yeah, just wondering whether it was used (esp. since is should prefer `jsonEncode` variant).

**Comment by jcollins-g** [comment]
OK, with the version pinned, this is ready for review.

File: pubspec.yaml [hunk 4]
```diff
@@ -4,14 +4,15 @@ description: The server backend for a web based interactive Dart service.
...
 
 dependencies:
   analysis_server_lib: ^0.1.2+2
   appengine: 0.4.0-flex.alpha.0+3
   archive: '>=1.0.19 <1.1.0'
   args: '>=0.12.0 <0.14.0'
   crypto: ^2.0.0
+  front_end: 0.1.0-alpha.4.1
```
Comment (devoncarew) [inline pubspec.yaml]: I don't think we need this import.

Comment (devoncarew) [reply pubspec.yaml]: And, it looks like `lib/src/pub.dart` has an undeclared dependency on `package:analyzer`.

Comment (jcollins-g) [reply pubspec.yaml]: It's actually an indirect dependency on something reexported from front_end, hence the pinning of a version here.

File: Dockerfile (hunk 1 above)
Comment (athomas) [reply Dockerfile]: They are built with every release. Whoever creates the release, needs to build & upload them. I did that for .58.

**Pull request merged**